upgraded some packages

Eldar2021 opened the pull request:
Upgraded some packages with `flutter pub upgrade` command help

clangenb reviewed (changes requested):
Nice, thanks for the maintenance. However, whenever you update the dev dependencies like mobx, and jsonserializable, make sure that the build runner still works:

The upgrade broke the build runner this time:

```

[INFO] Precompiling build script......
[WARNING] /C:/tools/flutter/.pub-cache/hosted/pub.dartlang.org/json_serializable_type_helper_utils-5.1.1/lib/src/custom_map_type_helper.dart:167:73: Error: A value of type 'DefaultContainer' can't be assigned to a variable of type 'String'.
 - 'DefaultContainer' is from 'package:json_serializable/src/default_container.dart' ('/C:/tools/flutter/.pub-cache/hosted/pub.dartlang.org/json_serializable-6.5.1/lib/src/default_container.dart').
      keyUsage = toFromString.deserialize(keyArg, keyUsage, false, true)!;
                                                                        ^
[INFO] Precompiling build script... completed, took 1.3s

[SEVERE] Failed to precompile build script .dart_tool/build/entrypoint/build.dart.
This is likely caused by a misconfigured builder definition.
```

Eldar2021 commented:
I fixed this error.
```

[INFO] Precompiling build script......
[WARNING] /C:/tools/flutter/.pub-cache/hosted/pub.dartlang.org/json_serializable_type_helper_utils-5.1.1/lib/src/custom_map_type_helper.dart:167:73: Error: A value of type 'DefaultContainer' can't be assigned to a variable of type 'String'.
 - 'DefaultContainer' is from 'package:json_serializable/src/default_container.dart' ('/C:/tools/flutter/.pub-cache/hosted/pub.dartlang.org/json_serializable-6.5.1/lib/src/default_container.dart').
      keyUsage = toFromString.deserialize(keyArg, keyUsage, false, true)!;
                                                                        ^
[INFO] Precompiling build script... completed, took 1.3s

[SEVERE] Failed to precompile build script .dart_tool/build/entrypoint/build.dart.
This is likely caused by a misconfigured builder definition.
```
And I deleted `^` special character for use package special version see more [stackoverflow](https://stackoverflow.com/questions/62248873/how-to-force-flutter-to-use-a-old-version-of-dependency) and [dart documentation](https://dart.dev/tools/pub/dependencies)

clangenb reviewed (approved):
Nice, I can confirm that the build runner works now. I have recreated the output and formatted it. The diff to the previous *.g files is zero, so I think it is safe to merge.
